ember-phone-input/src/components/phone-input.ts

```diff
@@ -152,73 +226,38 @@ export default class PhoneInputComponent extends Component {
    * @argument separateDialCode
    * @type {boolean}
    */
-  get separateDialCode() {
+  get separateDialCode(): boolean {
     return this.args.separateDialCode || false;
   }
 
-  _iti = null;
-
-  constructor() {
-    super(...arguments);
-
-    /**
-     * You have to implement this function to update the `number`.
-     * @argument update
-     * @param {string} number The international phoneNumber
-     * @param {Object} metadata The phoneNumber metadata
-     * @param {string} metadata.extension The extension part of the current number, so if the number was '+1 (702) 123-1234 ext. 12345' this would return '12345'.
-     * @param {Object} metadata.selectedCountryData The country data for the currently selected flag.
-     * @param {boolean} metadata.isValidNumber The validity of the current `phoneNumber`.
-     */
-
-    this.update = this.args.update || function () {};
-
-    if (this.customPlaceholder) {
-      assert(
-        '`customPlaceholder` must be of type string',
-        typeof this.customPlaceholder === 'string'
-      );
-    }
-
-    const validAutoPlaceholder = ['polite', 'aggressive', 'off'].includes(
-      this.autoPlaceholder
-    );
-
-    assert(
-      "`autoPlaceholder` possible values are 'polite', 'aggressive' and 'off'",
-      validAutoPlaceholder
-    );
-  }
-
   @action
-  onInput(event) {
+  onInput(event?: Event): boolean {
     const internationalPhoneNumber =
-      this._iti?.getNumber() ?? event?.target.value;
+      this._iti?.getNumber() ?? (event?.target as HTMLInputElement).value;
 
-    var meta = this._metaData(this._iti);
+    const meta = this._metaData(this._iti);
     this.update(internationalPhoneNumber, meta);
 
     return true;
   }
 
   @action
-  onDidUpdate() {
+  onDidUpdate(): void {
     this._formatNumber();
   }
 
   @action
-  onDidInsert(element) {
-    this.element = element;
-    this._loadAndSetup();
+  onDidInsert(element: HTMLInputElement): void {
+    this._loadAndSetup(element);
   }
 
   @action
-  onDestroy(element) {
+  onDestroy(element: HTMLInputElement): void {
     this._iti?.destroy();
     element.removeEventListener('countrychange', this._onCountryChange);
   }
 
-  async _loadAndSetup() {
+  async _loadAndSetup(element: HTMLInputElement): Promise<void> {
```

Comment:
Here applies a similar comment as the one I wrote here: https://github.com/qonto/ember-phone-input/pull/749/files#diff-34e5f38d343c0fe320d293f192edc4edb296ad67bbfce63e4fc87af453fed6eaR348

Reply:
Fixed